### PR TITLE
Enhance Advice with source location

### LINKED ISSUE #1444: [FEATURE]: Enhance Advice records with source location
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Advice records do not currently hold information about the location of the source code from which they are generated

### Proposed Solution

Add the missing information

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Enhance Advice with source type and location

### Linked issues
Resolves #1444 

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [ ] manually tested
- [x] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

**Comment by ericvergnaud** [comment]
I'll be adding tests as part of #1431 and #1439

File: src/databricks/labs/ucx/source_code/base.py [hunk 1]
```diff
@@ -15,8 +15,14 @@
...
 @dataclass
 class Advice:
+
+    MISSING_SOURCE_TYPE = "<MISSING_SOURCE_TYPE>"
+    MISSING_SOURCE_PATH = "<MISSING_SOURCE_PATH>"
+
     code: str
     message: str
+    source_type: str
+    source_path: str
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/base.py:25]: we should keep WHAT (e.g. deprecation) problem we have from WHERE (e.g. notebook or a file) that problem is. 

```
@dataclass
class NotebookAdvice:
  path: str
  advice: list[Advice]

@dataclass
class FileAdvice:
  path: str
  advice: list[Advice]
```

that being said, we may need another datastructure to map onto a database table, but it should have Job ID and Job Name as well and most likely live within the `assessment` package:

```
@dataclass
class WorkloadAdvice:
  workflow_id: int
  workflow_name: str
  object_type: string
  object_path: string # alternatively, object_id, to make links easier
  code: str
  message: str
  line: int
```

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/source_code/base.py (hunk 1 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/base.py:25]: See my comment in #1448

**Review by nfx — CHANGES_REQUESTED** [review]
do not modify `Advice`.

**Comment by ericvergnaud** [comment]
superseded by #1529

**Pull request closed without merging**